api/repositories/service_instance_repository.go

```diff
@@ -134,6 +143,56 @@ func (r *ServiceInstanceRepo) ListServiceInstances(ctx context.Context, authInfo
 	return returnServiceInstanceList(orderedServiceInstances), nil
 }
 
+func (r *ServiceInstanceRepo) GetServiceInstance(ctx context.Context, authInfo authorization.Info, guid string) (ServiceInstanceRecord, error) {
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return ServiceInstanceRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	serviceInstanceList := &servicesv1alpha1.CFServiceInstanceList{}
+	err = r.privilegedClient.List(ctx, serviceInstanceList, client.MatchingFields{"metadata.name": guid})
+	if err != nil {
+		return ServiceInstanceRecord{}, fmt.Errorf("failed to list service instances on cluster: %w", err)
+	}
+
+	record, err := returnServiceInstance(serviceInstanceList.Items)
+	if err != nil {
+		return ServiceInstanceRecord{}, err
+	}
+
+	err = userClient.Get(ctx, client.ObjectKey{Namespace: record.SpaceGUID, Name: record.GUID}, &servicesv1alpha1.CFServiceInstance{})
+	if k8serrors.IsForbidden(err) {
+		return ServiceInstanceRecord{}, NewForbiddenError(err)
+	}
```

Comment:
This is an example of what we discussed a few weeks back right? Since we don't know what K8s namespace the instance lives in we have to list so we're optimizing by doing that with the privileged client since it is able to maintain a cache.

I think that's fine and good, but I think the way we're doing it here is risky since we're actually using the return value of the privileged client call and only using the userClient for authz.

I think it would be safer if we refactored the privileged client actions into a function (like `privilegedLookupNamespaceForServiceInstance` (I'm sure y'all can come up with a better name 😅 )) that only returns the namespace and then we rely on the userClient for actually fetching the returned object.

Reply:
I know there's test that verify the insufficient permissions case, but I think this will both provide additional context for what is going on (via the function name) and give us a little more safety in the future.

Reply:
+1 for returning a record for the object being found by the unprivileged client